src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/reporting/generator/BedUtilisationReportGenerator.kt

```diff
@@ -26,7 +28,11 @@ class BedUtilisationReportGenerator(
 
   override val convert: BedEntity.(properties: BedUtilisationReportProperties) -> List<BedUtilisationReportRow> = { properties ->
     val startOfMonth = LocalDate.of(properties.year, properties.month, 1)
-    val endOfMonth = LocalDate.of(properties.year, properties.month, startOfMonth.month.length(startOfMonth.isLeapYear))
+    var endOfMonth = LocalDate.of(properties.year, properties.month, startOfMonth.month.length(startOfMonth.isLeapYear))
```

Comment:
could you please use a val here and assign using if/else to avoid value mutation

Reply:
We cant use if/else here because the endmonth has to referenced in DB query. The purpose of var is to change the value of the variable and i couldnt see any issue in using it in this context.